caddyhttp/httpserver/middleware.go

```diff
@@ -48,20 +47,48 @@ type (
 
 	// RequestMatcher checks to see if current request should be handled
 	// by underlying handler.
-	//
-	// TODO The long term plan is to get all middleware implement this
-	// interface and have validation done before requests are dispatched
-	// to each middleware.
 	RequestMatcher interface {
 		Match(r *http.Request) bool
 	}
+
+	// Config is a middleware configuration.
+	// This makes it possible for middlewares to have a common
+	// configuration interface.
+	//
+	// TODO The long term plan is to get all middleware implement this
+	// interface for configurations.
+	Config interface {
+		RequestMatcher
+		BasePath() string
```

Comment:
Hmm, if we can eliminate this method, then we may not even need the whole interface and can just use RequestMatcher.